osu.Game/Rulesets/Difficulty/Skills/Skill.cs

```diff
@@ -16,7 +16,12 @@ public abstract class Skill
         /// <summary>
         /// <see cref="DifficultyHitObject"/>s that were processed previously. They can affect the strain values of the following objects.
         /// </summary>
-        protected readonly LimitedCapacityStack<DifficultyHitObject> Previous = new LimitedCapacityStack<DifficultyHitObject>(2); // Contained objects not used yet
+        protected readonly ReverseQueue<DifficultyHitObject> Previous;
```

Comment:
I think it's okay for now? Unsure if interfacing is really worth it - the previous implementation was also open to such manipulation. Calling `Dequeue` isn't going to break the internals of `Skill` as is - only the user's own implementation.

Reply:
I'm happy either way. I don't think its a likely enough thing to need to protect against, but if we are concerned with API friendliness then its probably not a bad idea since it's trivial to implement a simple interface for the property.